portfolio/src/main/webapp/map.js

```diff
@@ -337,10 +338,60 @@ function postMarker(lat, lng, content) {
 /**
  * Tells the server to delete an individual marker. 
  */
-function deleteMarker(id) {
+function deleteMarker(marker, id, email) {
   if (window.confirm(DELETE_MARKER_WARNING)) {
     const params = new URLSearchParams();
     params.append('id', id);
-    fetch('/delete-marker', {method: 'POST', body: params});
+    params.append('email', email);
+    fetch('/delete-marker', {method: 'POST', body: params})
+    .then(response => response.json())
+    .then((status) => {
+      if (status.success === false) {
+        window.alert(DELETE_MARKER_FAIL);
+      } else {
+        marker.setMap(null);
+      }
+    });
+  }
+}
+
+/*
+ * Fetch the authentication status of the user from the server.
+ */
+function mapAuth() {
```

Comment:
My suggestion to avoid the duplicate code, is to abstract out the authentication portion of this and from script.js into a function in its own file. That function could take in the name of the element to which the login/logout text will be appended to along with an anonymous function that you can then call for the corresponding toggles i.e 
`
authenticateUser('map-container', function() { return toggleClickListener(logingStatus); });
`

Reply:
That makes sense. I actually wasn't aware that you could just call functions in between files which makes everything a lot easier! I didn't exactly use this solution but something similar to it.